scripts/cache-storm-data

```diff
@@ -0,0 +1,145 @@
+        try:
+            resp = requests.get(url, timeout=timeout)
+            resp.raise_for_status()
+            return resp.json()
+        except:
+            raise self.ConnectionFailure(url, timeout)
+
+    def is_detail_task_id(self, config, task_id):
+        for regex in config.executor_details_whitelist:
+            if regex.match(task_id):
+                return True
+        return False
+
+    def collect_detail_components(self, config, details):
+        detail_ids=[]
+        for spout in details.get('spouts'):
+            if self.is_detail_task_id(config, spout['spoutId']):
+                detail_ids.append(spout['encodedSpoutId'])
+        for bolt in details.get('bolts'):
+            name = bolt['boltId']
+            if self.is_detail_task_id(config, bolt['boltId']):
+                detail_ids.append(bolt['encodedBoltId'])
+        return detail_ids
+
+
+    def cache_one(self, config):
+        try:
+            cluster = self._get_json_from_url(config, '/api/v1/cluster/summary')
+            supervisors = self._get_json_from_url(config, '/api/v1/supervisor/summary')
+            all_topologies = self._get_json_from_url(config, '/api/v1/topology/summary')
+            topologies = storm_utils.collect_topologies(config.topologies, all_topologies.get('topologies'))
+            topology_url = urlparse.urljoin(config.url, '/api/v1/topology/')
+            topology_details = []
+            for name, topology in topologies.iteritems():
+                this_topology_url = urlparse.urljoin(topology_url, topology['id'])
+                details = self._get_json_from_url(config, this_topology_url, timeout=config.topology_timeout)
+                executor_components = self.collect_detail_components(config, details)
+                executor_details = []
+                for component in executor_components:
+                    detail_url = urlparse.urljoin(urlparse.urljoin(this_topology_url + '/', 'component/'), component)
+                    executor_details.append(self._get_json_from_url(config, detail_url, timeout=config.timeout))
+                topology_details.append({
+                    'name': name,
+                    'topology': details,
+                    'component_details': executor_details,
+                })
+
+            return {
+                'status': 'success',
+                'updated': time.time(),
+                'data': {
+                    'cluster': cluster,
+                    'supervisors': supervisors,
+                    'topologies': topologies,
+                    'topology_details': topology_details,
+                },
+            }
+        except self.ConnectionFailure as failure:
+            return {
+                'status': 'error',
+                'error_url': failure.url,
+                'error_timeout': failure.timeout,
+                'updated': time.time(),
+            }
+
+    def run(self):
+        for instance in self.instances:
```

Comment:
A good idea - I thinkkk we should make our own cron wrapper do that; I don't think I want to hardcode a metrics-collection scheme here (tempting as it would be).